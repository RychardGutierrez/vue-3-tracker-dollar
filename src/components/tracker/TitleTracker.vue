<template>
  <div class="flex flex-col items-center text-dollar-blue py-12">
    <div class="flex gap-7">
      <h1 class="text-5xl mb-2">{{ official }}</h1>
      <img :src="flagContry" :alt="nameDefaultFlag" class="w-24" />
    </div>
    <p class="text-base mb-2">{{ trackerDate }} - {{ trackerTime }}</p>
  </div>
</template>

<script setup>
defineProps({
  official: { typeof: String, require: true },
  flagContry: { typeof: String, require: true },
  nameDefaultFlag: { typeof: String, require: true },
  trackerDate: { typeof: String, require: true },
  trackerTime: { typeof: String, require: true },
})


</script>