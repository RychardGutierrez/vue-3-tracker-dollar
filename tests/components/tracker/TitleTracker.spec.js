import { shallowMount } from '@vue/test-utils';

import TitleTracker from '../../../src/components/tracker/TitleTracker.vue';

describe('TitleTracker component', () => {
  test('should to match with snapshot', () => {
    const props = {
      official: 'Plurinational State of Bolivia',
      flagContry: 'https://flagcdn.com/bo.svg',
      nameDefaultFlag: 'bolivia - Flag Icon',
      trackerDate: 'Sat, February 11',
      trackerTime: '8:20 PM',
    };

    const wrapper = shallowMount(TitleTracker, { props });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
