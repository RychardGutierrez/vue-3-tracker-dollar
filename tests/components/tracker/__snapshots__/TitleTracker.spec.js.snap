// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleTracker component should to match with snapshot 1`] = `
"<div class=\\"flex flex-col items-center text-dollar-blue py-12\\">
  <div class=\\"flex gap-7\\">
    <h1 class=\\"text-5xl mb-2\\">Plurinational State of Bolivia</h1><img src=\\"https://flagcdn.com/bo.svg\\" alt=\\"bolivia - Flag Icon\\" class=\\"w-24\\">
  </div>
  <p class=\\"text-base mb-2\\">Sat, February 11 - 8:20 PM</p>
</div>"
`;
